manifests/check.pp

```diff
@@ -236,10 +238,53 @@
     publish             => $publish,
     dependencies        => $dependencies,
     custom              => $custom,
+    content             => $content,
     subdue              => $subdue,
     proxy_requests      => $proxy_requests,
-    require             => File["${conf_dir}/checks"],
-    notify              => $::sensu::check_notify,
     ttl                 => $ttl,
   }
+
+  # Produce a new Hash map suitable for use with sensu::write_json.  The mapping
+  # behavior of Puppet parameters to scoped Sensu configuration attributes is
+  # implemented as a function.
+  $content_hash = sensu_check_content($resource_hash)
```

Comment:
@ghoneycutt I'd love to get your early feedback on this approach of using a parser function to create the output hash for sensu::write_json using the existing parameters of the defined type as input.